sentry_sdk/utils.py

```diff
@@ -1273,6 +1273,35 @@ class ServerlessTimeoutWarning(Exception):  # noqa: N818
     pass
 
 
+class SentryGraphQLClientError(Exception):
+    """Synthetic exception for GraphQL client errors."""
+
+    pass
+
+
+def _get_graphql_operation_name(query):
+    # type: (Dict[str, Any]) -> str
+    if query.get("operationName"):
+        return query["operationName"]
+
+    query = query["query"].strip()
+
+    match = re.match(r"^[a-z]* *([a-zA-Z0-9]+) \(?.*\)? *\{", query)
+    if match:
+        return match.group(1)
+    return "anonymous"
+
+
+def _get_graphql_operation_type(query):
+    # type: (Dict[str, Any]) -> str
+    query = query["query"].strip()
+    if query.startswith("mutation"):
```

Comment:
also maybe making it case insensitive? (and subsription below)

Reply:
done